Lunar/Data/Display.swift

```diff
@@ -119,11 +132,35 @@ class Display: NSManagedObject {
                         contrast = cap(newContrast.uint8Value, minVal: self.minContrast.uint8Value, maxVal: self.maxContrast.uint8Value)
                     }
                     _ = DDC.setContrast(for: self.id, contrast: contrast)
+					var currentValue = change.oldValue!.uint8Value
+					let incrementing = currentValue < contrast
+					let stepValue = abs(currentValue.distance(to: contrast)) >= 50 ? 2 : 1
+					func step () {
+						currentValue = self.incrOrDecrValue(incrementing, currentValue, stepValue, targetValue: contrast)
+						_ = DDC.setContrast(for: self.id, contrast: currentValue)
+						if currentValue == contrast {
+							return
+						}
+						DispatchQueue.main.asyncAfter(deadline: DispatchTime.now()) {
+							step()
+						}
+					}
+					step()
                     log.debug("\(self.name): Set contrast to \(contrast)")
                 }
             }),
         ]
     }
+	
+	func incrOrDecrValue (_ incrementing: Bool, _ currentValue: UInt8, _ stepValue: Int, targetValue: UInt8) -> UInt8 {
```

Comment:
It's better if we declare separate functions for increasing and decreasing. It makes the code clearer and it allows the compiler to skip branch prediction.